modules/data.atmosphere/R/met.process.R

```diff
@@ -234,30 +249,47 @@ met.process <- function(site, input_met, start_date, end_date, model,
-                                          str_ns = str_ns,
-                                          site = site, 
-                                          start_date = start_date, end_date = end_date, 
-                                          browndog = browndog, 
-                                          new.site = new.site,
-                                          overwrite = overwrite$met2model,
-                                          exact.dates = reg.model$exact.dates,
-                                          spin = spin)
-    
-    model.id  <- met2model.result$model.id
-    model.file.info <- db.query(paste0("SELECT * from dbfiles where id = ", model.id$dbfile.id), con)
-    model.file <- file.path(model.file.info$file_path,model.file.info$file_name)
-    
+    if(is.null(ready.id)){
+      model.file <- .met2model.local.module(inputfiles = inputfiles, 
+                                            model = model, 
+                                            con = con,
+                                            host = host, 
+                                            dir = dir, 
+                                            met = met, 
+                                            str_ns = str_ns,
+                                            site = site, 
+                                            start_date = start_date,
+                                            end_date = end_date, 
+                                            browndog = browndog, 
+                                            new.site = new.site,
+                                            overwrite = overwrite$met2model,
+                                            exact.dates = reg.model$exact.dates,
+                                            spin = spin)
+      
+    } else{
+      ## Get Model Registration
+      reg.model.xml <- system.file(paste0("register.", model, ".xml"), package = paste0("PEcAn.",model))
+      reg.model <- XML::xmlToList(XML::xmlParse(reg.model.xml))
+      
+      met2model.result <- .met2model.module(ready.id = ready.id, 
+                                            model = model, 
+                                            con = con,
+                                            host = host, 
+                                            dir = dir, 
+                                            met = met, 
+                                            str_ns = str_ns,
+                                            site = site, 
+                                            start_date = start_date, end_date = end_date, 
+                                            browndog = browndog, 
+                                            new.site = new.site,
+                                            overwrite = overwrite$met2model,
+                                            exact.dates = reg.model$exact.dates,
+                                            spin = spin)
+      
+      model.id  <- met2model.result$model.id
```

Comment:
is this section not needed later in the code?

Reply:
used in the next line